airflow/api_connexion/endpoints/dag_run_endpoint.py

```diff
@@ -158,26 +158,29 @@ def _apply_date_filters_to_query(
     return query
 
 
-@security.requires_authentication
+@security.requires_access([("can_read", "Dag"), ("can_read", "DagRun")])
 @provide_session
 def get_dag_runs_batch(session):
     """
     Get list of DAG Runs
     """
     body = request.get_json()
-    try:
+    try:  # TODO: Handle filtering.
         data = dagruns_batch_form_schema.load(body)
     except ValidationError as err:
         raise BadRequest(detail=str(err.messages))
 
+    appbuilder = current_app.appbuilder
+    readable_dag_ids = appbuilder.sm.get_readable_dag_ids(g.user)
     query = session.query(DagRun)
-
-    if data["dag_ids"]:
-        query = query.filter(DagRun.dag_id.in_(data["dag_ids"]))
+    if data.get("dag_ids"):
+        dag_ids = set(data["dag_ids"]) & set(readable_dag_ids)
+        query = query.filter(DagRun.dag_id.in_(dag_ids))
+    else:
+        query = query.filter(DagRun.dag_id.in_(readable_dag_ids))
 
     dag_runs, total_entries = _fetch_dag_runs(
         query,
-        session,
```

Comment:
What was the reason for removing the session here?

Reply:
It has to do with how the count is getting made, as seen in your previous comment. It used to use `session`, but it doesn't need to now.